packages/odf/components/s3-browser/upload-objects/upload-component/FileUploadComponent.tsx

```diff
@@ -17,146 +21,148 @@ import './fileUploadComponent.scss';
-    | UploadStatus.UPLOAD_COMPLETE
-    | UploadStatus.UPLOAD_START
-  >(UploadStatus.INIT_STATE);
+export const FileUploadComponent: React.FC<FileUploadComponentProps> = observer(
+  ({
+    client,
+    bucketName,
+    showSidebar,
+    setCompletionTime,
+    triggerRefresh,
+    uploadStore,
+  }) => {
+    const { t } = useCustomTranslation();
+    const [uploadStatus, setUploadStatus] = React.useState<
+      | UploadStatus.INIT_STATE
+      | UploadStatus.UPLOAD_COMPLETE
+      | UploadStatus.UPLOAD_START
+    >(UploadStatus.INIT_STATE);
 
-  const [searchParams] = useSearchParams();
-  const foldersPath = searchParams.get(PREFIX) || '';
+    const [searchParams] = useSearchParams();
+    const foldersPath = searchParams.get(PREFIX) || '';
 
-  const processFiles = React.useCallback(
-    async (uploadObjects: File[], setProgress) => {
-      try {
-        const completionTime = await uploadFile(
-          uploadObjects,
-          client,
-          bucketName,
-          setProgress,
-          foldersPath
-        );
-        setCompletionTime(completionTime);
-      } catch (e) {
-        // eslint-disable-next-line no-console
-        console.error('Error uploading file', e);
-      }
-    },
-    [bucketName, client, foldersPath, setCompletionTime]
-  );
+    const processFiles = React.useCallback(
+      async (uploadObjects: File[]) => {
+        try {
+          const completionTime = await uploadFile(
+            uploadObjects,
+            client,
+            bucketName,
+            foldersPath,
+            uploadStore
+          );
+          setCompletionTime(completionTime);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error('Error uploading file', e);
+        }
+      },
+      [bucketName, client, foldersPath, setCompletionTime, uploadStore]
+    );
 
-  const onDrop = React.useCallback(
-    async (acceptedFiles: File[]) => {
-      const intialUploadObjects = acceptedFiles.reduce((acc, curr) => {
-        const name = getPrefix(
-          curr.webkitRelativePath || curr.name,
-          foldersPath
-        );
-        acc[name] = convertFileToUploadProgress(curr);
-        return acc;
-      }, {} as UploadProgress);
-      setUploadProgress(intialUploadObjects);
-      setUploadStatus(UploadStatus.UPLOAD_START);
-      await processFiles(acceptedFiles, setUploadProgress);
-      setUploadStatus(UploadStatus.UPLOAD_COMPLETE);
-      triggerRefresh();
-    },
-    [setUploadProgress, processFiles, triggerRefresh, foldersPath]
-  );
+    const onDrop = React.useCallback(
+      async (acceptedFiles: File[]) => {
+        acceptedFiles.forEach((file) => {
+          const key = getPrefix(
+            file.webkitRelativePath || file.name,
+            foldersPath
+          );
+          uploadStore.addFile(convertFileToUploadProgress(file, key), key);
+        });
+        setUploadStatus(UploadStatus.UPLOAD_START);
+        const batches = _.chunk(acceptedFiles, 8);
```

Comment:
can we make it `6` instead ?? given that's the typical browser limit per domain: https://medium.com/@hnasr/chromes-6-tcp-connections-limit-c199fe550af6 
```suggestion
        const batches = _.chunk(acceptedFiles, 6);
```

Reply:
`8` is also fine, but just asking...